Python unit test failure in test_style_transfer on macOS 10.13

```
_____________________ StyleTransferTest.test_export_coreml _____________________

self = <turicreate.test.test_style_transfer.StyleTransferTest testMethod=test_export_coreml>

    def test_export_coreml(self):
        import coremltools
        filename = tempfile.mkstemp('my_style_transfer.mlmodel')[1]
        model = self.model
>       model.export_coreml(filename)

test_style_transfer.py:214: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../toolkits/style_transfer/style_transfer.py:843: in export_coreml
    mlmodel = coremltools.models.MLModel(spec)
../../coremltools/models/model.py:209: in __init__
    self.__proxy__ = _get_proxy_from_spec(filename)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

filename = '/var/folders/yb/4phfy8f539ng78zzgw90b6sh0000gn/T/tmpU60fLm.mlmodel'

    def _get_proxy_from_spec(filename):
        try:
            from ..libcoremlpython import _MLModelProxy
        except:
            _MLModelProxy = None
    
        if _MLModelProxy:
            # check if the version is supported
            engineVersion = _MLModelProxy.maximum_supported_specification_version()
            spec = _load_spec(filename)
            if spec.specificationVersion > engineVersion:
                # in this case the specification is a newer kind of .mlmodel than this version of the engine can support
                # so we'll not try to have a proxy object
                return None
            # check if there are custom layers
            if _has_custom_layer(spec):
                # custom layers can't be supported directly by compiling and loading the model here
                return None
    
>           return _MLModelProxy(filename)
E           RuntimeError: Error compiling model: "Error reading protobuf spec. validator error: The .mlmodel supplied is of version 3, intended for a newer version of Xcode. This version of Xcode supports model version 2 or earlier.".

../../coremltools/models/model.py:103: RuntimeError
```

This seems to be due to https://github.com/apple/coremltools/issues/198 -- until we can take a dependency on a newer coremltools version, we should work around this in the test so it can pass on 10.13.

--- a/src/unity/python/turicreate/toolkits/style_transfer/style_transfer.py
+++ b/src/unity/python/turicreate/toolkits/style_transfer/style_transfer.py
@@ -840,14 +840,14 @@ class StyleTransfer(_CustomModel):
         flexible_shape_utils.update_image_size_range(spec, feature_name=self.content_feature, size_range=img_size_ranges)
         flexible_shape_utils.update_image_size_range(spec, feature_name=stylized_image, size_range=img_size_ranges)
 
-        mlmodel = coremltools.models.MLModel(spec)
         model_type = 'style transfer (%s)' % self.model
-        mlmodel.short_description = _coreml_utils._mlmodel_short_description(
+        spec.description.metadata.shortDescription = _coreml_utils._mlmodel_short_description(
             model_type)
-        mlmodel.input_description[self.content_feature] = 'Input image'
-        mlmodel.input_description['index'] = u'Style index array (set index I to 1.0 to enable Ith style)'
-        mlmodel.output_description[stylized_image] = 'Stylized image'
-        _coreml_utils._set_model_metadata(mlmodel, self.__class__.__name__, {
+        spec.description.input[0].shortDescription = 'Input image'
+        spec.description.input[1].shortDescription = u'Style index array (set index I to 1.0 to enable Ith style)'
+        spec.description.output[0].shortDescription = 'Stylized image'
+        user_defined_metadata = _coreml_utils._get_model_metadata(
+            self.__class__.__name__, {
                 'model': self.model,
                 'num_styles': str(self.num_styles),
                 'content_feature': self.content_feature,
@@ -855,7 +855,9 @@ class StyleTransfer(_CustomModel):
                 'max_iterations': str(self.max_iterations),
                 'training_iterations': str(self.training_iterations),
             }, version=StyleTransfer._PYTHON_STYLE_TRANSFER_VERSION)
-        mlmodel.save(path)
+        spec.description.metadata.userDefined.update(user_defined_metadata)
+        from coremltools.models.utils import save_spec as _save_spec
+        _save_spec(spec, path)
 
     def get_styles(self, style=None):
         """

